zerver/lib/streams.py

```diff
@@ -146,10 +146,19 @@ def create_streams_if_needed(
     added_streams: List[Stream] = []
     existing_streams: List[Stream] = []
     for stream_dict in stream_dicts:
+        is_invite_only = stream_dict.get("invite_only", False)
+        if acting_user is not None:
+            if (
+                is_invite_only
+                and not acting_user.can_create_private_streams()
+                or not is_invite_only
+                and not acting_user.can_create_public_streams()
+            ):
+                raise JsonableError(_("Insufficient permission"))
```

Comment:
I'd prefer this as two if statements rather than an `or` statement -- that way `coverage` will check that we test both code paths.